examples/horizontal_convection.jl

```diff
@@ -0,0 +1,356 @@
+    ζ_snapshot = file["timeseries/ζ/$iter"][4:end-3, 1, 4:end-3]
+    χ_snapshot = file["timeseries/χ/$iter"][4:end-3, 1, 4:end-3]
+    
+    ## determine colorbar limits and contour levels
+    slim = 0.6
+    slevels = vcat(range(0, stop=slim, length=31), [0.6])
+
+    blim = 0.6
+    blevels = vcat([-1], range(-blim, stop=blim, length=51), [1])
+    
+    ζlim = 9
+    ζlevels = nice_divergent_levels(ζ_snapshot, ζlim)
+
+    χlim = 0.025
+    χlevels = nice_levels(χ_snapshot, χlim)
+
+    @info @sprintf("Drawing frame %d from iteration %d:", i, iter)
+
+    kwargs = (      xlims = (-Lx/2, Lx/2),
+                    ylims = (-H, 0),
+                   xlabel = "x / H",
+                   ylabel = "z / H",
+              aspectratio = 1,
+                linewidth = 0)
+
+    s_plot = contourf(xs, zs, s_snapshot';
+                      clims = (0, slim), levels = slevels,
+                      color = :speed, kwargs...)
+    s_title = @sprintf("speed √[(u²+w²)/(b⋆H)] @ t=%1.2f", t)
+    
+    b_plot = contourf(xb, zb, b_snapshot';
+                      clims = (-blim, blim), levels = blevels,
+                      color = :thermal, kwargs...)
+    b_title = @sprintf("buoyancy, b/b⋆ @ t=%1.2f", t)
+    
+    ζ_plot = contourf(xζ, zζ, ζ_snapshot';
+                      clims=(-ζlim, ζlim), levels = ζlevels,
+                      color = :balance, kwargs...)
+    ζ_title = @sprintf("vorticity, (∂u/∂z - ∂w/∂x) √(H/b⋆) @ t=%1.2f", t)
+    
+    χ_plot = contourf(xχ, zχ, χ_snapshot';
+                      clims = (0, χlim), levels = χlevels,
+                      color = :dense, kwargs...)
+    χ_title = @sprintf("buoyancy dissipation, κ|∇b|² √(H/b⋆⁵) @ t=%1.2f", t)
+    
+    plot(s_plot, b_plot, ζ_plot, χ_plot,
+            dpi = 120,
+           size = (700.25, 1200.25),
+           link = :x,
+         layout = Plots.grid(4, 1),
+          title = [s_title, b_title, ζ_title, χ_title])
+
+    iter == iterations[end] && close(file)
+end
+
+mp4(anim, "horizontal_convection.mp4", fps = 16) # hide
+
+# ### The Nusselt number
+#
+# We are often interested on how much the flow enhances the mixing. This is quantified by the
+# Nusselt number, which measure how much the fluid flow enhances mixing occurs compared if only
+# diffusion was in operation. The Nusselt number is given by
+#
+# ```math
+# Nu = \frac{\langle \chi \rangle}{\langle \chi_{\rm diff} \rangle} \, ,
+# ```
+#
+# where angle brackets above denote a spatial and time average and  ``\chi_{\rm diff}`` is
+# the buoyancy dissipation without any flow, i.e., it's the buoyancy dissipation associated
+# with the buoyancy distribution that satisfies
+#
+# ```math
+# \kappa \nabla^2 b_{\rm diff} = 0 \, ,
+# ```
+#
+# with the same boundary conditions same as our setup.
+
+χ_timeseries = FieldTimeSeries("horizontal_convection.jld2", "χ")
+nx, ny, nz, nt = size(χ_timeseries)
+
+Nu = reshape(sum(χ_timeseries, dims=(1, 2, 3))*grid.Δx*grid.Δz, (nt,))
```

Comment:
I load `χ` as a timeseries with `FieldTimeSeries` and then I wanna compute the domain integral of this. Is there a functionality to do so? I didn't find any so I did it using `sum()`... any better ideas?

cc @ali-ramadhan, @glwagner

Reply:
oh, hidden in the source code I saw `Oceananigans.AbstractOperations.volume` which might be of interest here... Why is this not in the docs? Can I multiply this with a `FieldTimeSeries`? When I tried the time-dimension seems to disappeared...

Reply:
It's not in the docs because there's no example or experiment using it. It's intended to be used with `IntegratedField` (and we need its cousin for correct `AveragedField` on stretched grids). We can implement that for this example if you plan to use it.

Reply:
`IntegratedField`! That sounds like what I want. Will look into that -- thanks!